script/20240516-revoke-roles/02_Revoke_Roles.s.sol

```diff
@@ -0,0 +1,175 @@
+import { INSDomainPrice } from "src/interfaces/INSDomainPrice.sol";
+
+contract Migration__01_Revoke_Roles is Migration {
+  using Strings for *;
+  using LibRNSDomain for string;
+
+  address duke = 0x0F68eDBE14C8f68481771016d7E2871d6a35DE11;
+  address multisig = 0x1FF1edE0242317b8C4229fC59E64DD93952019ef;
+
+  RNSUnified internal _rns;
+  RNSAuction internal _auction;
+  NameChecker internal _nameChecker;
+  RNSDomainPrice internal _domainPrice;
+  PublicResolver internal _publicResolver;
+  RNSReverseRegistrar internal _reverseRegistrar;
+  RONRegistrarController internal _ronController;
+  OwnedMulticaller internal _ownedMulticaller;
+
+  function run() external onlyOn(DefaultNetwork.RoninMainnet.key()) {
+    _rns = RNSUnified(loadContract(Contract.RNSUnified.key()));
+    _auction = RNSAuction(loadContract(Contract.RNSAuction.key()));
+    _nameChecker = NameChecker(loadContract(Contract.NameChecker.key()));
+    _domainPrice = RNSDomainPrice(loadContract(Contract.RNSDomainPrice.key()));
+    _publicResolver = PublicResolver(loadContract(Contract.PublicResolver.key()));
+    _reverseRegistrar = RNSReverseRegistrar(loadContract(Contract.RNSReverseRegistrar.key()));
+    _ronController = RONRegistrarController(loadContract(Contract.RONRegistrarController.key()));
+    _ownedMulticaller = OwnedMulticaller(loadContract(Contract.OwnedMulticaller.key()));
+
+    address[] memory contracts = new address[](5);
+    contracts[0] = address(_domainPrice);
+    contracts[1] = address(_ronController);
+    contracts[2] = address(_nameChecker);
+    contracts[3] = address(_rns);
+    contracts[4] = address(_auction);
+
+    vm.startBroadcast(duke);
+
+    // Transfer .ron domain ownership to owned multicaller
+    _rns.transferFrom(duke, address(_ownedMulticaller), 0x0);
+    _rns.transferFrom(duke, address(_ownedMulticaller), LibRNSDomain.RON_ID);
+    _rns.transferFrom(duke, address(_ownedMulticaller), LibRNSDomain.ADDR_REVERSE_ID);
+
+    address[] memory tos = new address[](3);
+    bytes[] memory callDatas = new bytes[](3);
+    uint256[] memory values = new uint256[](3);
+
+    tos[0] = address(_rns);
+    tos[1] = address(_rns);
+    tos[2] = address(_rns);
+
+    callDatas[0] = abi.encodeCall(ERC721.setApprovalForAll, (address(_auction), true));
+    callDatas[1] = abi.encodeCall(ERC721.setApprovalForAll, (address(_ronController), true));
+    callDatas[2] = abi.encodeCall(ERC721.approve, (address(_reverseRegistrar), LibRNSDomain.ADDR_REVERSE_ID));
+
+    values[0] = 0;
+    values[1] = 0;
+    values[2] = 0;
+
+    _ownedMulticaller.multicall(tos, callDatas, values);
+
+    uint256 length = contracts.length;
+
+    for (uint256 i; i < length; i++) {
+      AccessControlEnumerable(contracts[i]).grantRole(0x0, multisig);
+      console.log("Duke will renounce his admin roles of contract:", vm.getLabel(contracts[i]), "manually");
+
+      assertTrue(
+        AccessControlEnumerable(contracts[i]).getRoleMemberCount(0x0) > 0,
```

Comment:
As current admin will renounce his role manually, I think the assertion should be:
```suggestion
        AccessControlEnumerable(contracts[i]).getRoleMemberCount(0x0) > 1,
```
We also have another admin on domain price contract, please consider to remove it as well. Ref https://sky-mavis.slack.com/archives/C06C3HW1HS7/p1712812933009569

Reply:
updated.